sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -398,27 +398,173 @@ class SQLContext(@transient val sparkContext: SparkContext)
     jsonRDD(json.rdd, samplingRatio);
   }
 
+  /**
+   * :: Experimental ::
+   * Loads a dataset from the given path as a DataFrame.
+   * It will use the default data source configured in spark.sql.sources.default.
```

Comment:
I'd just say: "Returns the dataset stored at `path` as a DataFrame, using the default datasource configured in spark.sql.sources.default."